alphagov/paas-release-ci
# [#170180331] Use Credhub for secrets

What
----
Uploads all secrets to Credhub, and has Concourse inject them at runtime.

How to review
-------------

1. Create a new build env from `paas-bootstrap` (you’ll need to add the new entries in Google API console, so that SSO works)
 
    ```make dev build-concourse bootstrap DEPLOY_ENV=test-ci ```

 2. Get your dev envs CF admin credentials, or otherwise create a user for the deployer

 3. Upload the cf-cli secrets in `paas-release-ci` to your new build env, from this branch

     make dev credhub DEPLOY_ENV=test-ci 

     make dev upload-cf-cli-secrets \
     DEPLOY_ENV=test-ci \
    CF_USER=user \
    CF_PASSWORD=password

    make dev upload-rubbernecker-secrets

4. Upload the other secrets

```
    make dev credhub DEPLOY_ENV=test-ci
     make dev upload-aiven-secrets 
    make dev upload-rubbernecker-secrets
    make dev upload-zendesk-secrets
     make dev upload-hackmd-secrets  
```

 5. Set the `paas-release-ci` pipelines from this branch

     make dev pipelines \
     CF_DEPLOY_ENV=andyhunt \
    DEPLOY_ENV=test-ci \
    BRANCH=use_credhub_for_secrets_170180331

  6. Kick off the setup pipeline
7. Run the `rubbernecker`, `paas-product-page`, `hackmd` pipelines and check they run successfully
8. To test the `aiven-broker` pipeline, push an empty commit on [the related aiven broker PR](https://github.com/alphagov/paas-aiven-broker/pull/23) if it doesn't trigger.

After review
----
Make sure you change [this line in `scripts/integration-test-pipelines.sh`](https://github.com/alphagov/paas-release-ci/pull/136/files#diff-6055aa4b63249226553f0f063b72801eR65) to ensure that `aiven-broker` uses the `master` branch again.

Who can review
--------------
Anyone
